intera_interface/src/intera_interface/limb.py

```diff
@@ -519,3 +588,147 @@ def test_collision():
+          # If empty, the default gain of 0.4 will be used
+          ikreq.nullspace_gain.append(nullspace_gain)
+
+        try:
+            resp = self._iksvc(ikreq)
+        except (rospy.ServiceException, rospy.ROSException), e:
+            rospy.logerr("IK Service call failed: %s" % (e,))
+            return False
+        limb_joints = {}
+        # Check if result valid, and type of seed ultimately used to get solution
+        if (resp.result_type[0] > 0):
+            seed_str = {
+                        ikreq.SEED_USER: 'User Provided Seed',
+                        ikreq.SEED_CURRENT: 'Current Joint Angles',
+                        ikreq.SEED_NS_MAP: 'Nullspace Setpoints',
+                       }.get(resp.result_type[0], 'None')
+            # Format solution into Limb API-compatible dictionary
+            limb_joints = dict(zip(resp.joints[0].name, resp.joints[0].position))
+        else:
+            rospy.logerr("INVALID POSE - No Valid Joint Solution Found.")
+            return False
+        return limb_joints
+
+    def fk_request(self, joint_angles,
+                        end_point='right_hand'):
+        """
+        Forward Kinematics request sent to FK Service
+
+        @type joint_angles: [double]
+        @param joint_angles: the arm's joint positions
+        @type end_point: string
+        @param end_point: name of the end point (should be in URDF)
+        @return: Forward Kinematics response from FK service
+        """
+        fkreq = SolvePositionFKRequest()
+        # Add desired pose for forward kinematics
+        joints = JointState()
+        joints.name = self._joint_names
+        joints.position = joint_angles
+        fkreq.configuration.append(joints)
+        # Request forward kinematics from base to end_point
+        fkreq.tip_names.append(end_point)
+        try:
+            resp = self._fksvc(fkreq)
+        except (rospy.ServiceException, rospy.ROSException), e:
+            rospy.logerr("FK Service call failed: %s" % (e,))
+            return False
+        return resp
+
+    def joint_angles_to_cartesian_pose(self, joint_angles,
+                                       end_point='right_hand'):
+        """
+        Convert joint angles to a Cartesian Pose. Calls FK service
+
+        @type joint_angles: [double]
+        @param joint_angles: the arm's joint positions
+        @type end_point: string
+        @param end_point: name of the end point (should be in URDF)
+        @type pose: geometry_msgs.Pose
+        @return: cartesian pose associated with joint state
+        """
+        resp = self.fk_request(joint_angles, end_point)
+
+        if resp is None:
+            rospy.logerr("Failed to get cartesian pose - check FK server.")
+            return None
+        if not resp.isValid[0]:
+            rospy.loginfo("INVALID JOINTS - No Cartesian Solution Found.")
+            return None
+        return resp.pose_stamp[0].pose
+
+    def in_collision(self, joint_angles,
```

Comment:
I can rename the function if that's clearer. 

The INVALID JOINTS error will only print if the user calls both joint_angles_to_cartesian_pose and in_collision in a row without checking the first's output. The better solution would be the user calling fk_request and parsing the full return message.